integr8ly/integreatly-operator
# 3scale update to 0.5.2

# Description


Upgrade 3scale operator to 0.5.2 to test component patch release process. https://issues.redhat.com/browse/INTLY-7936



## Type of change





## Checklist

## Verification steps

### Verify fresh install from PR branch
- create operator namespace:
```sh
oc new-project redhat-rhmi-operator
```
- prepare secrets on cluster:
```sh
make cluster/prepare/smtp && make cluster/prepare/pagerduty && make cluster/prepare/dms
```
- create CatalogSource for install:
```
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: rhmi-operators
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/damurphy/integreatly-index:2.3.1
```
- UI -> Operators -> Operator Hub -> RHMI Operator -> Install (ns: redhat-rhmi-operator, approval: manual)
- UI -> Home -> Search -> InstallPlan -> approve 
- Monitor RHMI CR until last stage (Solution Explorer) installed:
```sh
watch -d 'oc get rhmi rhmi -o yaml | grep "completed" -B 4 -A 1'
```
- set up idp: 
```sh
./scripts/setup-sso-idp.sh
```
- run functional tests to verify cluster + 3scale functionality:
```sh
make test/functional
```
- check 3scale product consoles
- verify 3scale-Openshift integration, steps [here](https://github.com/integr8ly/integreatly-operator/blob/master/test-cases/tests/products/h03-verify-integration-between-3scale-and-openshift.md)

### Verify upgrade from 2.3.0 to PR changes
- create operator namespace:
```sh
oc new-project redhat-rhmi-operator
```
- prepare secrets on cluster:
```sh
make cluster/prepare/smtp && make cluster/prepare/pagerduty && make cluster/prepare/dms
```
- create CatalogSource for install:
```
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: rhmi-operators
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/damurphy/integreatly-index:2.3.0
```
- UI -> Operators -> Operator Hub -> RHMI Operator -> Install (ns: redhat-rhmi-operator, approval: manual)
- UI -> Home -> Search -> InstallPlan -> approve 
- Monitor RHMI CR until last stage (Solution Explorer) installed:
```sh
watch -d 'oc get rhmi rhmi -o yaml | grep "completed" -B 4 -A 1'
```
- set up idp:
```sh
./scripts/setup-sso-idp.sh
```
- upgrade RHMI to version with 0.5.2:
```sh 
oc edit catalogsource rhmi-operators -n openshift-marketplace
  image: quay.io/damurphy/integreatly-index:2.3.1
```
- run functional tests to verify cluster + 3scale functionality:
```sh
make test/functional
```
- check 3scale product consoles
- verify 3scale-Openshift integration, steps [here](https://github.com/integr8ly/integreatly-operator/blob/master/test-cases/tests/products/h03-verify-integration-between-3scale-and-openshift.md)
